src/widgets/component-new-edit/components/variables/calculated-variables.jsx

```diff
@@ -39,9 +42,17 @@ export const defaultProps = {
   errors: [],
 };
 
-// Component
-
-function CalculatedVariables({ formName, selectorPath, errors, addErrors }) {
+function CalculatedVariables({ formName, selectorPath, errors, addErrors, componentsStore }) {
+  const scopeOption = Object.values(componentsStore)
+  .filter(component=> component.type === "LOOP" && !component.basedOn || component.type === "LOOP" && component.basedOn && componentsStore[component.basedOn] && componentsStore[component.basedOn].type != "LOOP" )
```

Comment:
you should use a const for the string "LOOP"
This line is way too long. you should first wrap your tests between ( ) because for exemple you have twice this test : `component.type === "LOOP"`.